### ISSUE TITLE
Incorrect CLI usage flags for forest and forest-cli

### ISSUE THREAD

**Issue by LesnyRumcajs** [ChainSafe/forest#2076]
**Describe the bug**

After the binaries split was introduced in https://github.com/ChainSafe/forest/issues/1901, some of the flags don't make any sense for some of the binaries, e.g. `--import-snapshot` for `forest-cli` or `--token` for `forest` (unless I'm missing something)

**To Reproduce**
1. Run `forest-cli`
2. See usage flags.

**Log output**

<details>
  <summary>Log Output</summary>
  
  ```Paste log output here
❯ target/debug/forest-cli --help
forest-cli 0.4.1+git.110e83ad
ChainSafe Systems <info@chainsafe.io>
Forest binary sub-commands available

USAGE:
    forest-cli [FLAGS] [OPTIONS] <SUBCOMMAND>

FLAGS:
        --detach               Daemonize Forest process
        --halt-after-import    Halt with exit code 0 after successfully importing a snapshot
    -h, --help                 Prints help information
        --skip-load            Skips loading CAR file and uses header to index chain. Assumes a pre-loaded database
    -V, --version              Prints version information

OPTIONS:
        --chain <chain>
            Choose network chain to sync to [default: mainnet]  [possible values: mainnet, calibnet]

    -c, --config <config>                            A TOML file containing relevant configurations
        --encrypt-keystore <encrypt-keystore>        Encrypt the key-store (default: true)
    -g, --genesis <genesis>                          The genesis CAR file
        --height <height>                            Validate snapshot at given EPOCH
        --import-chain <import-chain>                Import a chain from a local CAR file or URL
        --import-snapshot <import-snapshot>          Import a snapshot from a local CAR file or URL
    -k, --kademlia <kademlia>                        Allow Kademlia (default: true)
        --mdns <mdns>                                Allow MDNS (default: false)
        --metrics-address <metrics-address>
            Address used for metrics collection server. By defaults binds on localhost on port 6116

        --req-window <req-window>                    Number of tipsets requested over chain exchange (default is 200)
    -r, --rpc <rpc>                                  Allow RPC to be active or not (default: true)
        --rpc-address <rpc-address>                  Address used for RPC. By defaults binds on localhost on port 1234
        --target-peer-count <target-peer-count>      Amount of Peers we want to be connected to (default is 75)
        --tipset-sample-size <tipset-sample-size>
            Number of tipsets to include in the sample that determines what the network head is

    -t, --token <token>                              Client JWT token to use for JSON-RPC authentication

SUBCOMMANDS:
    auth            Manage RPC permissions
    chain           Interact with Filecoin blockchain
    config          Manage node configuration
    fetch-params    Download parameters for generating and verifying proofs for given size
    genesis         Work with blockchain genesis
    help            Prints this message or the help of the given subcommand(s)
    mpool           Interact with the message pool
    net             Manage P2P network
    send            Send funds between accounts
    snapshot        Manage snapshots
    state           Interact with and query Filecoin chain state
    sync            Inspect or interact with the chain synchronizer
    wallet          Manage wallet

  ```
</details> 

**Expected behaviour**

Flags available for `forest` and `forest-cli` are only those that make sense and are usable. There should not be any flags that belong solely to another binary.

**Screenshots**

**Environment (please complete the following information):**
 - OS: Fedora 36
 - Rust version(e.g. `rustc --version`) `rustc 1.66.0-nightly (470e518c4 2022-09-27)`
 - Branch/commit 110e83ad (`main`)

**Other information and links**

**Comment by hanabi1224** [comment]
The options are kept mostly because the cli and daemon share the logic of config file, as per @lemmih the config file is only needed for the daemon, a refactor to remove config logic from cli would help fix this

**Comment by LesnyRumcajs** [comment]
@jdjaustin Please note one caveat - currently we have a single command to dump the default TOML config - `forest-cli config dump`. This mostly dumps the daemon configuration (a bit counterintuitive). A slightly more ergonomic approach would be to move it to `forest config dump` and optionally `forest-cli config dump` if we want to support config files for the CLI binary.

**Comment by jdjaustin** [comment]
@LesnyRumcajs do you think we need to keep this issue open? Seems like it's a bit out of date.

**Comment by LesnyRumcajs** [comment]
It is still relevant. `forest-cli` exposes a bunch of flags it doesn't really use. On the other hand, we can assume it's superseded by https://github.com/ChainSafe/forest/issues/3172. @lemmih ?

**Comment by lemmih** [comment]
Hmm, let's keep this issue open and use it as a minder to check the flags after #3172 is closed.

**Comment by aatifsyed** [comment]
I have some ideas about this, I can build a PoC for the team

**Comment by lemmih** [comment]
@LesnyRumcajs The flags look good to me now.

```
USAGE:
  forest-cli [OPTIONS] <COMMAND>

SUBCOMMANDS:
  chain     Interact with Filecoin blockchain
  auth      Manage RPC permissions
  net       Manage P2P network
  sync      Inspect or interact with the chain synchronizer
  mpool     Interact with the message pool
  state     Interact with and query Filecoin chain state
  config    Manage node configuration
  snapshot  Manage snapshots
  send      Send funds between accounts
  info      Print node info
  attach    Attach to daemon via a JavaScript console
  shutdown  Shutdown Forest
  help      Print this message or the help of the given subcommand(s)

OPTIONS:
  -t, --token <TOKEN>  Client JWT token to use for JSON-RPC authentication
  -h, --help           Print help
  -V, --version        Print version
```

Could you confirm and close this issue if everything is okay?

**Comment by LesnyRumcajs** [comment]
There's a small thing I'll resolve in #3958, otherwise LGTM.
